pkg/nodeconfig/nodeconfig.go

```diff
@@ -129,6 +129,43 @@ func discoverKubeAPIServerEndpoint() (string, error) {
 	return host.Status.APIServerInternalURL, nil
 }
 
+// getWICDCredentials returns the CA cert and access token associated with the WICD service account
+func getWICDCredentials() (*windows.Authentication, error) {
+	cfg, err := crclientcfg.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get config to talk to kubernetes api server")
+	}
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating kubernetes clientset")
+	}
+	operatorNamespace := "openshift-windows-machine-config-operator"
+	secrets, err := clientset.CoreV1().Secrets(operatorNamespace).List(context.TODO(), meta.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error listing secrets in namespace %s", operatorNamespace)
+	}
+
+	tokenSecretPrefix := "windows-instance-config-daemon-token-"
+	var filteredSecrets []core.Secret
+	for _, secret := range secrets.Items {
+		if strings.HasPrefix(secret.Name, tokenSecretPrefix) {
+			filteredSecrets = append(filteredSecrets, secret)
+		}
+	}
+	if len(filteredSecrets) != 1 {
+		return nil, fmt.Errorf("expected 1 secret with '%s' prefix, found %d", tokenSecretPrefix, len(filteredSecrets))
+	}
+	caCert := filteredSecrets[0].Data[core.ServiceAccountRootCAKey]
+	if len(caCert) == 0 {
```

Comment:
What if `core.ServiceAccountRootCAKey` is not present?

Reply:
Then caCert would be initialized to an empty byte array (len == 0) so this if statement would catch it.

Reply:
It would help debugging if we can differentiate between the two cases.